### PR TITLE
feat: Adds RNN support to controlflow

### LINKED ISSUE #54: Add RNN Language Model
Add an LSTM language model using Wikitext2 data

### PR DESCRIPTION
**Pull request by Panaetius**
Adds RNN support in our controlflow, supporting hidden state

Closes #54

### CODE REVIEW

File: docs/dataset.rst [hunk 1]
```diff
@@ -82,3 +82,22 @@ DatasetCifar
 
 .. autoclass:: DatasetCifar
     :members:
+
+
+NLP
+~~~~~~~~~~~~~~~~
+
+.. automodule:: mlbench_core.dataset.nlp
```
Comment (ehoelzl) [inline docs/dataset.rst:90]: Use `autoapimodule`, `autoapiclass` and `autoapifunction` so that the docs works without installing mlbench-core

**Comment by martinjaggi** [comment]
can we close this?

**Comment by Panaetius** [comment]
@martinjaggi  Yes we can merge it, just needs a review/approval

File: mlbench_core/dataset/nlp/pytorch/language_model.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import math
```
Comment (ehoelzl) [inline mlbench_core/dataset/nlp/pytorch/language_model.py:1]: Isn't it less confusing if we name the dataloader classes like the dataset they represent (i.e. `wikitext2_dataset.py`), as in other benchmarks ? Cause we now have 3 different datasets (including transformer) in `nlp/`

File: docs/dataset.rst [hunk 3]
```diff
@@ -81,3 +81,22 @@ DatasetCifar
...
+.. currentmodule:: mlbench_core.dataset.nlp
+
+pytorch
++++++++
+
+.. automodule:: mlbench_core.dataset.nlp.pytorch
```
Comment (ehoelzl) [inline docs/dataset.rst:95]: Can you change this file's directives to use `autoapi`, like in `models.rst` ?

Comment (Panaetius) [reply docs/dataset.rst:95]: good catch, I think this snuck back in when I rebased on develop :+1:

**Pull request merged**